spec/models/membership_application_spec.rb

```diff
@@ -140,22 +149,27 @@
 
   end
 
-  describe "#destroy" do
+  describe 'destroy callbacks' do
 
-    let(:application_owner) {create(:user, email: 'user_1@random.com')}
-    let(:uploaded_file) {create(:uploaded_file, actual_file: (File.new(File.join(FIXTURE_DIR, 'image.jpg'))))}
-    let(:membership_application) {create(:membership_application, user: application_owner, uploaded_files: [uploaded_file])}
+    let(:application_owner) { create(:user, email: 'user_1@random.com') }
+
+    app_file = (File.new(File.join(FIXTURE_DIR, 'image.jpg')))
+    let(:uploaded_file) { create(:uploaded_file, actual_file: app_file) }
 
-    it 'destroys a membershipapplication' do
-      membership_application.destroy
-      expect(membership_application.destroyed?).to be_truthy
+    let(:application) do
+      create(:membership_application, user: application_owner,
+             uploaded_files: [uploaded_file], state: :accepted)
     end
 
-    it 'destroys the uploaded file' do
-      membership_application.destroy
+    it 'invokes method to delete uploaded files' do
+      application.destroy
       expect(uploaded_file.destroyed?).to be_truthy
     end
 
+    it "destroys associated company" do
+      expect(application.company).to receive(:destroy)
+      application.destroy
+    end
   end
```

Comment:
Need to test both branches of the conditional:
```
  company.destroy if (company.membership_applications.count == 0)
```

You've tested the  _if (company.membership_applications.count == 0)_  branch (possibility) above. 

You also need a test to for the branch  _if (company.membership_applications.count != 0)_  to ensure that the company is _not_ destroyed

Here's how I'd write it:
```
it "does not destroy the company if the company has at least 1 other application" do
      the_company = application.company
      create(:membership_application, user: create(:user),
                           company_number: the_company.company_number,
                           state: :accepted)

      expect(the_company.membership_applications.count).to eq 2

      expect(application.company).not_to receive(:destroy)
      application.destroy
      expect(the_company.membership_applications.count).to be 1
    end
```